### PR TITLE
add adler32 functionality and tests

### PR DESCRIPTION
**Pull request by maarten-k**
exposes the adler32 function of libdeflate.  Implementation is almost a copy of the crc32 implementation, only the initial value is 1 in adler32 (where crc32 has 0).

### CODE REVIEW

File: tests/test_adler32.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        b"foobar",
+        b"foobar" * 42,
+        os.urandom(50),
+    ],
+)
+def test_adler32_with_start_value(data):
+    """test whether adler32 function computes correctly,
+    with start value (reference: zlib.adler32)"""
+    # 0 is the default start value
+    assert deflate.adler32(data, 1) == deflate.adler32(data, 1)
```
Comment (ThomasWaldmann) [inline tests/test_adler32.py:42]: this looks strange.

Comment (ThomasWaldmann) [inline tests/test_adler32.py]: - the ", 1" in the right call needs to be removed.
- the value in the comment needs to be corrected (1, not 0)

the idea of this test is to check whether the result with the start value explicitly given (but of same value as the default value) is the same as when the start value is not given.

**Comment by ThomasWaldmann** [comment]
While you're at it, you could also correct the typo "alder32" in the commit comment.

**Comment by ThomasWaldmann** [comment]
Everything else looks pretty good and as expected, thanks for the PR!

**Comment by maarten-k** [comment]
Thanks for the review and thanks for spotting the mistake with the assert.

File: tests/test_adler32.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        b"foobar",
+        b"foobar" * 42,
+        os.urandom(50),
+    ],
+)
+def test_adler32_with_start_value(data):
+    """test whether adler32 function computes correctly,
+    with start value (reference: zlib.adler32)"""
+    # 1 is the default start value
+    assert deflate.adler32(data,1) == deflate.adler32(data)
```
Comment (ThomasWaldmann) [inline tests/test_adler32.py]: nitpick: add a blank after the comma or some pep8 checker will complain later...

**Review by ThomasWaldmann — APPROVED** [review]
LGTM!

@dcwatson maybe squash commits together when you merge into master?

**Pull request merged**